sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -326,4 +369,256 @@ void autoRenewLockOnReceiveMessage() {
             receiver.close();
         }
     }
+
+    @Test
+    void testBasicReceiveAndDeleteWithBinaryData() {
+        // Arrange
+        final String messageTrackingId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = TestUtils.getServiceBusMessageBinary(messageTrackingId, 0, 64);
+
+        // Assert & Act
+        StepVerifier.create(sender.send(message).thenMany(receiveDeleteModeReceiver.receive()))
+            .assertNext(receivedMessage ->
+                Assertions.assertTrue(receivedMessage.getProperties().containsKey(MESSAGE_TRACKING_ID)))
+            .expectNoEvent(Duration.ofSeconds(2))
+            .thenCancel()
+            .verify();
+    }
+
+    @Test
+    void testBasicReceiveAndCompleteWithLargeBinaryData() {
+        // Arrange
+        final String messageTrackingId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = TestUtils.getServiceBusMessageBinary(messageTrackingId, 0, 64 * 1024);
+
+        // Assert & Act
+        StepVerifier.create(sender.send(message).thenMany(receiveDeleteModeReceiver.receive()))
+            .assertNext(receivedMessage ->
+                Assertions.assertTrue(receivedMessage.getProperties().containsKey(MESSAGE_TRACKING_ID)))
+            .expectNoEvent(Duration.ofSeconds(2))
+            .thenCancel()
+            .verify();
+    }
+
+    @Test
+    void testBasicReceiveAndComplete() {
+        // Arrange
+        final String messageTrackingId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = TestUtils.getServiceBusMessageBinary(messageTrackingId, 0, 64 * 1024);
+
+        final ServiceBusReceivedMessage receivedMessage = sender.send(message)
+            .then(receiverManualComplete.receive().next())
+            .block(Duration.ofSeconds(30));
+
+        Assertions.assertNotNull(receivedMessage);
+
+        // Assert & Act
+        StepVerifier.create(receiverManualComplete.complete(receivedMessage))
+            .verifyComplete();
+    }
+
+    @Test
+    void testBasicReceiveAndCompleteMessageWithProperties() {
```

Comment:
This is covered by testBasicReceiveAndDeleteWithBinaryData plus we always assert we get properties back.